dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/sam/SamPermissionsImpl.java

```diff
@@ -321,11 +321,11 @@ private void addPolicy(ResourcesApi resourcesApi, String resourceId, String poli
 
     @Override
     public boolean canDoAction(User user, Workflow workflow, Role.Action action) {
-        ResourcesApi resourcesApi = getResourcesApi(user);
-        String encodedPath = encodedWorkflowResource(workflow, resourcesApi.getApiClient());
         try {
```

Comment:
You could have a `if hasGoogleToken(user)` at the top to short-circuit things, like a couple of other methods do.